### ISSUE TITLE
PDF generation on div linux systems fails with versions >1.5.59 

### ISSUE THREAD

**Issue by tpurschke** [HakanL/WkHtmlToPdf-DotNet#72]
I know this issue has come up numerous times before but I see a common behaviour here across all tested deb-based linux platforms only depending on the WkHtmlToPdf-DotNet version which is used.

We tested PDF generation on ubuntu 20.04, debian 10, debian 11 and all have the same behaviour:

working fine with v1.5.59

    <PackageReference Include="Haukcode.WkHtmlToPdfDotNet" Version="1.5.59" />

but when upgrading to 1.5.68 (without any other changes), the PDF generation fails with the usual message "Unable to load native library. The platform may be missing native dependencies (libjpeg62, etc)"

    <PackageReference Include="Haukcode.WkHtmlToPdfDotNet" Version="1.5.68" />

Example packages on Ubuntu 20.04:
```
tim@fworch-comp:~/tpurschke-fwo$ dpkg -l | grep jpeg
ii  libjpeg-turbo8:amd64                       2.0.3-0ubuntu1.20.04.1                amd64        IJG JPEG compliant runtime library.
ii  libjpeg-turbo8-dev:amd64                   2.0.3-0ubuntu1.20.04.1                amd64        Development files for the IJG JPEG library
ii  libjpeg62:amd64                            1:6b2-3                               amd64        Independent JPEG Group's JPEG runtime library (version 6.2)
ii  libjpeg8:amd64                             8c-2ubuntu8                           amd64        Independent JPEG Group's JPEG runtime library (dependency package)
ii  libjpeg8-dev:amd64                         8c-2ubuntu8                           amd64        Independent JPEG Group's JPEG runtime library (dependency package)
tim@fworch-comp:~/tpurschke-fwo$ 
```

Also tested with additional installation of
wkhtmltopdf libgdiplus libc6-dev
--> to no avail

**Comment by tpurschke** [comment]
sorry - wrong repo :-(

**Comment by HakanL** [comment]
Can you reproduce this in docker? It would be easier to figure it out that way.

**Comment by tpurschke** [comment]
cannot reproduce the issue 
- with our test project
- with a separate mini project
- when debugging UI in vscode under Windows or Ubuntu 20.04
- even a scheduled PDF report (via middleware server) works fine on the server itself
 
in all of the above cases everything works fine with 1.5.68

Only interactively creating a PDF report via the UI (https://github.com/CactuseSecurity/firewall-orchestrator) server running on a remote server fails.
After downgrading to 1.5.59 the PDF generation via UI also works again.

This could point to a timing issue as root cause. But still no clue where this might come from.

@HakanL do you have any idea what might have an influence in versions after 1.5.59 around timing?

**Comment by HakanL** [comment]
Hmm, that is odd, the only real change in the later version is how we load the library, I can't remember exactly what the change was about, but I think it was to support the old .NET Framework. Make sure you only have a global SynchronizedConverter in your project (instantiated at startup typically).

**Comment by tpurschke** [comment]
finally found the solution:

    dotnet publish --no-self-contained -c {{ dotnet_mode }} -o {{ ui_start_dir }}/bin/{{ dotnet_mode }}/net{{ dotnet_version }} -p:PublishSingleFile=true -r linux-x64

This does not work anymore for versions >1.5.59
when removing the "-p:PublishSingleFile=true" and "-r linux-x64" switches, everything works as expected.

Maybe you can see the deeper reasoning behind this.

Thanks for your help Hakan and keep up the good work!

**Comment by HakanL** [comment]
@tpurschke Can you try this prerelease nuget package? https://www.nuget.org/packages/Haukcode.WkHtmlToPdfDotNet/1.5.69-test-linux-singlefile-233789a9

**Comment by tpurschke** [comment]
> Ha, interesting, I bet it's the PublishSingleFile that's causing it, it doesn't know how to load the native library. 
that's what we thought but it turned out that it did not work without the single file switch. Had to remove the "-r linux-x64" as well to get it working.

**Comment by HakanL** [comment]
Ah ok. But see if the prerelease nuget package works, I basically put the previous load attempt back (but after the existing code instead of before).

**Comment by tpurschke** [comment]
> @tpurschke Can you try this prerelease nuget package? https://www.nuget.org/packages/Haukcode.WkHtmlToPdfDotNet/1.5.69-test-linux-singlefile-233789a9

yep - was able to get it running with both switches with the above nuget package

**Comment by HakanL** [comment]
@tpurschke That's great news! I've merged it to master, version [1.5.71 on NuGet](https://www.nuget.org/packages/Haukcode.WkHtmlToPdfDotNet/1.5.71).

**Comment by tpurschke** [comment]
cheers Hakan!
